build/Microsoft.DotNet.Cli.Test.targets

```diff
@@ -0,0 +1,157 @@
+      <TestOutputDir>$(BaseOutputDirectory)/tests</TestOutputDir>
+      <TestPackagesDir>$(TestOutputDir)/packages</TestPackagesDir>
+      <TestArtifactsDir>$(TestOutputDir)/artifacts</TestArtifactsDir>
+      <TestResultXmlDir>$(TestOutputDir)/results</TestResultXmlDir>
+      <TestDirectory>$(RepoRoot)/test</TestDirectory>
+      <DotnetUnderTest>$(DotnetStage2)</DotnetUnderTest>
+    </PropertyGroup>    
+
+  </Target>
+
+  <Target Name="Test" 
+          DependsOnTargets="PrepareTests;
+                            SetupTests;
+                            RestoreTests;
+                            BuildTests;
+                            RunTests;" />
+
+  <Target Name="SetupTests"
+          DependsOnTargets="SetupTestProjects;" />
+
+  <Target Name="SetupTestProjects"
+          DependsOnTargets="SetupTestPackageProjectData;
+                            RestoreTestAssetProjects;
+                            RestoreDesktopTestAssetProjects;
+                            BuildTestAssetProjects;
+                            BuildDesktopTestAssetProjects;" />
+
+  <Target Name="SetupTestPackages" 
+          DependsOnTargets="RestoreTestAssetPackages;
+                            SetupTestPackageProjectData"
+          Outputs="%(TestPackageProject.Identity)">
+    <PropertyGroup>
+      <VersionSuffixOption></VersionSuffixOption>
+      <VersionSuffixOption Condition="'%(TestPackageProject.VersionSuffix)' != ''">--version-suffix %(TestPackageProject.VersionSuffix)</VersionSuffixOption>
+    </PropertyGroup>
+    <Exec Command="$(DotnetUnderTest) pack %(TestPackageProject.FullPath) --output $(TestPackagesDir) $(VersionSuffixOption)" />
+  </Target>
+
+  <Target Name="RestoreTestAssetPackages" 
+          DependsOnTargets="PrepareTests;
+                            SetupRestoreTestAssetPackagesInputs;" 
+          Inputs="@(RestoreTestAssetPackagesInputs)"
+          Outputs="@(RestoreTestAssetPackagesInputs->'%(RelativeDir)/project.lock.json')">
+    <Exec Command="$(DotnetUnderTest) restore $(RepoRoot)/TestAssets/TestPackages/" />
+  </Target>
+
+  <Target Name="SetupRestoreTestAssetPackagesInputs">
+    <ItemGroup>
+      <RestoreTestAssetPackagesInputs Include="$(RepoRoot)/TestAssets/TestPackages/**/project.json" />
+    </ItemGroup>
+  </Target>
+
+  <Target Name="RestoreTestAssetProjects"
+          DependsOnTargets="SetupTestPackages;
+                            SetupRestoreTestAssetProjectsInputs;"
+          Inputs="@(RestoreTestAssetProjectsInputs)"
+          Outputs="@(RestoreTestAssetProjectsInputs->'%(RelativeDir)/project.lock.json')">
+    <Exec Command="$(DotnetUnderTest) restore $(RepoRoot)/TestAssets/TestProjects/ --fallbacksource $(TestPackagesDir)" />
+  </Target>
+
+  <Target Name="SetupRestoreTestAssetProjectsInputs" >
+    <ItemGroup>
+      <RestoreTestAssetProjectsInputs Include="$(RepoRoot)/TestAssets/TestProjects/**/project.json" />
+    </ItemGroup>
+  </Target>
+
+  <Target Name="CleanTestAssetProjectsLockFiles" >
```

Comment:
I'm going to leave it for now, then. The context of the Creator targets is also interesting to have next to the cleanup targets. Once we have more of these it will be easier to tell how we manage them.